test/jdk/java/net/httpclient/ExpectContinueTest.java

```diff
@@ -0,0 +1,296 @@
+
+        // Due to limitations of the above Http1 Server, a manual approach is taken to test the hanging with the
+        // httpclient using Http1 so that the correct response header can be returned for the test case
+        http1HangServer = new Http1HangServer(saHang);
+        hangUri = URI.create("http://" + http1HangServer.ia.getCanonicalHostName() + ":" + http1HangServer.port + "/http1/hang");
+
+
+        http2TestServer = HttpTestServer.of(
+                new Http2TestServer("localhost", false, 0));
+        http2TestServer.addHandler(new GetHandler(), "/http2/get");
+        http2TestServer.addHandler(new PostHandler(), "/http2/post");
+        http2TestServer.addHandler(new PostHandlerCantContinue(), "/http2/hang");
+        h2getUri = URI.create("http://" + http2TestServer.serverAuthority() + "/http2/get");
+        h2postUri = URI.create("http://" + http2TestServer.serverAuthority() + "/http2/post");
+        h2hangUri = URI.create("http://" + http2TestServer.serverAuthority() + "/http2/hang");
+
+        http1TestServer.start();
+        http1HangServer.start();
+        http2TestServer.start();
+    }
+
+    @AfterTest
+    public void teardown() throws IOException {
+        http1TestServer.stop();
+        http1HangServer.close();
+        http2TestServer.stop();
+    }
+
+    static class GetHandler implements HttpTestHandler {
+
+        @Override
+        public void handle(HttpTestExchange exchange) throws IOException {
+            try (InputStream is = exchange.getRequestBody();
+                 OutputStream os = exchange.getResponseBody()) {
+                is.readAllBytes();
+                byte[] bytes = "RESPONSE_BODY".getBytes(UTF_8);
+                exchange.sendResponseHeaders(200, bytes.length);
+                os.write(bytes);
+            }
+        }
+    }
+
+    static class PostHandler implements HttpTestHandler {
+
+        @Override
+        public void handle(HttpTestExchange exchange) throws IOException {
+            // Http1 server has already sent 100 response at this point but not Http2 server
+            if (exchange.getExchangeVersion().equals(HttpClient.Version.HTTP_2)) {
+                // Send 100 Headers, tell client that we're ready for body
+                exchange.sendResponseHeaders(100, 0);
+            }
+
+            // Read body from client and acknowledge with 200
+            try (InputStream is = exchange.getRequestBody();
+                OutputStream os = exchange.getResponseBody()) {
+                is.readAllBytes();
+                exchange.sendResponseHeaders(200, 0);
+            }
+        }
+    }
+
+    static class PostHandlerCantContinue implements HttpTestHandler {
+
+        @Override
+        public void handle(HttpTestExchange exchange) throws IOException {
+            //Send 417 Headers, tell client to not send body
+            try (InputStream is = exchange.getRequestBody();
```

Comment:
Should we include a couple of other variants of the server to cover some other possible flows. For example:

- Don't read the request body, but just send the 417 response header?
- Send some other final response status code other than 100 or 417?

Reply:
The suggestion is merely to verify that we have covered the necessary code paths related to this flow. You can skip this, if looking at the code you see that the current test covers it.